### PR TITLE
Service test

### PR DESCRIPTION
**Pull request by Raychenko88**
Service test

### CODE REVIEW

File: src/test/java/info/sjd/service/impl/CartServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import info.sjd.model.Cart;
+import info.sjd.model.User;
+import info.sjd.service.CartService;
+import info.sjd.service.UserService;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+
+import javax.xml.crypto.Data;
```
Comment (Ignatenko2207) [inline src/test/java/info/sjd/service/impl/CartServiceImplTest.java]: Do you need it?

File: src/test/java/info/sjd/service/impl/CartServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    @Autowired
+    UserService userService;
+
+    @MockBean
+    CartDAO cartDAO;
+
+    @Test
+    void saveTest() {
+        User user = User.builder().login("test_login111").password("test_pass").firstName("test_fn").lastName("test_ln").build();
```
Comment (Ignatenko2207) [inline src/test/java/info/sjd/service/impl/CartServiceImplTest.java]: All fields in new lines like
.field(value)
.nextField(nextValue)
.build();

File: src/test/java/info/sjd/service/impl/CartServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        Cart cart = Cart.builder().closed(0).user(user).creationTime(currentTime).build();
```
Comment (Ignatenko2207) [inline src/test/java/info/sjd/service/impl/CartServiceImplTest.java]: The same style. Change it everywhere.

File: src/main/java/info/sjd/service/impl/UserServiceImpl.java [hunk 4]
```diff
@@ -14,7 +14,7 @@
 public class UserServiceImpl implements UserService {
 
     @Autowired
-    private UserDAO userDAO;
+    protected UserDAO userDAO;
```
Comment (Ignatenko2207) [inline src/main/java/info/sjd/service/impl/UserServiceImpl.java]: Access must be private. Change packages for test, but not access.

**Review by Ignatenko2207 — COMMENTED** [review]
Tests for services must be in package service, not in impl.

File: src/test/java/info/sjd/service/impl/CartServiceImplTest.java (hunk 1 above)
Comment (Raychenko88) [reply src/test/java/info/sjd/service/impl/CartServiceImplTest.java]: done

File: src/test/java/info/sjd/service/impl/CartServiceImplTest.java (hunk 2 above)
Comment (Raychenko88) [reply src/test/java/info/sjd/service/impl/CartServiceImplTest.java]: done

File: src/test/java/info/sjd/service/impl/CartServiceImplTest.java (hunk 3 above)
Comment (Raychenko88) [reply src/test/java/info/sjd/service/impl/CartServiceImplTest.java]: done

File: src/main/java/info/sjd/service/impl/UserServiceImpl.java (hunk 4 above)
Comment (Raychenko88) [reply src/main/java/info/sjd/service/impl/UserServiceImpl.java]: done

File: src/test/java/info/sjd/service/CartServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+import java.util.Date;
+
+import static java.util.Optional.of;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.Mockito.*;
+
+@SpringBootTest
+class CartServiceImplTest {
```
Comment (Ignatenko2207) [inline src/test/java/info/sjd/service/CartServiceImplTest.java]: Class name is not correct.

File: src/test/java/info/sjd/service/OrderServiceImplTest.java [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+import java.util.Date;
+
+import static java.util.Optional.of;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.Mockito.*;
+
+@SpringBootTest
+class OrderServiceImplTest {
```
Comment (Ignatenko2207) [inline src/test/java/info/sjd/service/OrderServiceImplTest.java]: Class name is not correct

File: src/test/java/info/sjd/service/CartServiceImplTest.java (hunk 5 above)
Comment (Raychenko88) [reply src/test/java/info/sjd/service/CartServiceImplTest.java]: done

File: src/test/java/info/sjd/service/OrderServiceImplTest.java (hunk 6 above)
Comment (Raychenko88) [reply src/test/java/info/sjd/service/OrderServiceImplTest.java]: done

**Review by Ignatenko2207 — APPROVED** [review]

**Pull request merged**